src/settings/settings.validation.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  PipeTransform,
+  Injectable,
+  ArgumentMetadata,
+  BadRequestException,
+} from '@nestjs/common';
+import { CreateSettingDto } from './dto/create-setting.dto';
+import { UpdateSettingDto } from './dto/update-setting.dto';
+
+@Injectable()
+export class SettingsBodyValidation implements PipeTransform {
+  validateValue(dataType: string, value: any): boolean {
+    if (dataType === 'string') {
+      return typeof value === 'string';
+    } else if (dataType === 'boolean') {
+      return value === true || value === false;
+    } else if (dataType === 'number') {
+      return typeof value === 'number';
+    } else if (dataType === 'object') {
+      return typeof value === 'object';
```

Comment:
The validation will pass if I pass null to the value property because null is also an object

Reply:
Acknowledged
Clarification: Are settings allowed to have null or undefined values? I have assumed that they are not values that you would ideally want in the DB.

Reply:
Settings are not allowed to have null/undefined values

Reply:
Done